### PR TITLE
[core][compiled graphs] Add ray[cg] pip install option

### PR DESCRIPTION
**Pull request by ruisearch42**
## Why are these changes needed?

Similar to https://github.com/ray-project/ray/pull/47009
ray[adag] is still used in vLLM, we will remove it after changing vLLM to use ray[cg]

## Related issue number

Part of https://github.com/ray-project/ray/issues/47934

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/setup.py [hunk 1]
```diff
@@ -235,6 +235,9 @@ def get_packages(self):
         "adag": [
             "cupy-cuda12x; sys_platform != 'darwin'",
         ],
+        "cg": [
+            "torch-cuda12x; sys_platform != 'darwin'",
```
Comment (aslonnie) [inline python/setup.py:239]:
```
$ pip install torch-cuda12x
ERROR: Could not find a version that satisfies the requirement torch-cuda12x (from versions: none)
ERROR: No matching distribution found for torch-cuda12x
```

there is no such pypi package on public pypi.org

Comment (aslonnie) [reply python/setup.py:239]: https://pypi.org/project/torch-cuda12x/ this page does not exist.

Comment (ruisearch42) [reply python/setup.py]: typo, fixed, thanks for catching

File: python/setup.py [hunk 2]
```diff
@@ -235,6 +235,9 @@ def get_packages(self):
         "adag": [
             "cupy-cuda12x; sys_platform != 'darwin'",
         ],
+        "cg": [
+            "cupy-cuda12x; sys_platform != 'darwin'",
+        ],
```
Comment (aslonnie) [inline python/setup.py:240]: could you add a comment saying `adag` is just an alias of `cg`?

**Review by aslonnie — APPROVED** [review]

File: python/setup.py (hunk 2 above)
Comment (aslonnie) [reply python/setup.py:240]: better if you just remove `adag` here, and have an `setup_spec.extras["adag"] = list(setup_spec.extras["cg"]` below

**Review by kevin85421 — APPROVED** [review]

**Review by aslonnie — APPROVED** [review]

**Pull request merged**